src/ext/java/org/opentripplanner/ext/flex/trip/ContinuousPickupDropOffTrip.java

```diff
@@ -0,0 +1,275 @@
+
+    // This would return only trips which can be found by raptor
+    if (isToStop) { return Stream.empty(); }
+
+    for (int fromIndex = (int) Math.floor(toIndex); fromIndex >= 0; fromIndex--) {
+      if (stopTimes[fromIndex].pickupType == PICKDROP_NONE) continue;
+      for (StopLocation stop : expandStops(stopTimes[fromIndex].stop)) {
+        res.add(new FlexEgressTemplate<>(egress, this, (double) fromIndex, toIndex, stop, date, continuousCalculator));
+      }
+    }
+
+    return res.stream();
+  }
+
+  @Override
+  public int earliestDepartureTime(
+      int departureTime, Double fromStopIndex, Double toStopIndex, int flexTime
+  ) {
+    if (Math.rint(fromStopIndex) == fromStopIndex) {
+      return stopTimes[(int) (double) fromStopIndex].departureTime;
+    }
+    int stopTime = getStopTime(fromStopIndex);
+    return stopTime >= departureTime ? stopTime : -1;
+  }
+
+  @Override
+  public int latestArrivalTime(
+      int arrivalTime, Double fromStopIndex, Double toStopIndex, int flexTime
+  ) {
+    if (Math.rint(toStopIndex) == toStopIndex) {
+      return stopTimes[(int) (double) toStopIndex].arrivalTime;
+    }
+    int stopTime = getStopTime(toStopIndex);
+    return stopTime <= arrivalTime ? stopTime : -1;
+  }
+
+  private int getStopTime(Double stopIndex) {
+    int prevStop = (int) Math.floor(stopIndex);
+    int nextStop = (int) Math.ceil(stopIndex);
+
+    int departureFromPrevious = stopTimes[prevStop].departureTime;
+    int arrivalToNext = stopTimes[nextStop].arrivalTime;
+
+    return departureFromPrevious + (int) ((arrivalToNext - departureFromPrevious) * (stopIndex - prevStop));
+  }
+
+  @Override
+  public Collection<StopLocation> getStops() {
+    List<StopLocation> stops = new ArrayList<>(Arrays.asList(continuousStops));
+    Arrays.stream(stopTimes).forEach(stopTime -> stops.add(stopTime.stop));
+
+    return stops;
+  }
+
+  @Override
+  public BookingInfo getBookingInfo(int i) {
+    return bookingInfos[i];
+  }
+
+  public void addGeometries(
+      Graph graph, StreetMatcher matcher
+  ) {
+    TripPattern pattern = graph.index.getPatternForTrip().get(trip);
+
+    if (pattern == null || pattern.getGeometry() == null) { return; }
+
+    FeedScopedId tripId = trip.getId();
+
+    ArrayList<Vertex> tempVertices = new ArrayList<>();
+    ArrayList<Double> tempStopIndices = new ArrayList<>();
+    ArrayList<Double> tempDistances = new ArrayList<>();
+
+    double cumulativeDistance = 0;
+
+    for (int i = 0; i < pattern.numHopGeometries(); i++) {
+      ContinuousPickupDropOffStopTime stopTime = stopTimes[i];
+      if (stopTime.continuousPickupType == PICKDROP_NONE
+          && stopTime.continuousDropOffType == PICKDROP_NONE) {
+        continue;
+      }
+
+      List<StreetEdge> edges = matcher.match(pattern.getHopGeometry(i));
+      if (edges == null || edges.isEmpty()) { continue; }
+
+      StopLocation stop = stopTime.stop;
+      // TODO: How to generate id?
```

Comment:
sounds fine? Or do we need to add `"_" + i` to make it unique?

Reply:
The generated StopLocation is the area between the two stops, so it should not have the stop id of the preceding stop

Reply:
Then we should use both IDs of both stops?